### PR TITLE
Add Flutter support to the FluentIcons library

### PR DESCRIPTION
**Pull request by aakash1313**
This PR is to add flutter support to the FluentIcons package
Issue: #34 

Steps to test:

Windows/Mac
1. Install Android Studio 
2. Install flutter for Windows/Mac : https://flutter.dev/docs/get-started/install
3. In Android Studio add Flutter plugin 
        a. Configure -> Plugin -> Search for flutter
4. Create a sample flutter application to make sure that everything is setup correctly.
5. Clone the project , switch to the flutter branch
6. open fluentui-system-icons as an existing Android project
7. For testing on Android  : ->Create an Android emulator or actual android device 
8. For testing on iOS: -> Create an iOS simulator , in your device section in Android studio , select the iOS simulator and run the app.

### CODE REVIEW

File: flutter/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## [0.0.1] - TODO: Add release date.
```
Comment (spencer-nelson) [inline flutter/CHANGELOG.md:1]: We should probably go ahead and fill this out, at least as far as we can

File: flutter/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+## Icon List
+
+[View the full list of icons](../icons.md)
+
+### Icon styles
+The library offers icons in two styles, `regular` and `filled`
+
+regular | filled
+--------|-------
+![ic_fluent_mail_regular](../art/ic_fluent_mail_regular.png)|![ic_fluent_mail_filled](../art/ic_fluent_mail_filled.png)
```
Comment (spencer-nelson) [inline flutter/README.md:18]: This brings up a great question. @jasoncuster do we have a "default" or representative icon you would suggest be used here?

File: README.md [hunk 3]
```diff
@@ -79,11 +91,16 @@ You can build and run the demo apps following the steps below.
 3. Select the `sample-showcase` in the build configuration dropdown
 4. Click run 
 
+### Flutter
+
+// TODO: @aakash1313
```
Comment (willhou) [inline README.md:96]: @aakash1313 could you fill this out please :)

File: flutter/CHANGELOG.md (hunk 1 above)
Comment (willhou) [reply flutter/CHANGELOG.md:1]: I think we could remove this file and use tag description to document any changes if needed. https://github.com/microsoft/fluentui-system-icons/releases/tag/1.1.42

File: flutter/README.md (hunk 2 above)
Comment (willhou) [reply flutter/README.md:18]: I think this is fine, but we could update it to use the SVGs from the assets folder since it's supported by Github markdown. Examples can be found in https://github.com/microsoft/fluentui-system-icons/blob/master/icons.md

File: flutter/example/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# flutter_sample_showcase
+
+A new Flutter project.
```
Comment (willhou) [inline flutter/example/README.md:3]: This REAME will needs to be updated.

Comment (willhou) [reply flutter/example/README.md:3]: Sorry, we should remove this

File: flutter/CHANGELOG.md (hunk 1 above)
Comment (aakash1313) [reply flutter/CHANGELOG.md]: @willhou : Addressed :)

File: README.md (hunk 3 above)
Comment (aakash1313) [reply README.md]: Addressed

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -62,6 +62,8 @@ jobs:
       run: python3 generate_build_gn_android.py
       working-directory: importer
 
+    ## Flutter - TODO
```
Comment (spencer-nelson) [inline .github/workflows/publish.yml:65]: If this is going to be left in as TODO, please add an issue for tracking/discussing it.

File: flutter/README.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
+# fluentui_system_icons
```
Comment (spencer-nelson) [inline flutter/README.md:1]: This should probably be stylized more readably as FluentUI System Icons

Comment (spencer-nelson) [reply flutter/README.md:1]: or just Fluent System Icons, as that's what the iOS/Android libraries call it

File: flutter/README.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      // Use the FluentIcons + name of the icon you want
+        icon: Icon(FluentIcons.access_time_24_regular),
+        onPressed: () { print("Button pressed"); }
+    );
+  }
+}
+```
+## Example
+
+View the Flutter app in the `example` directory to see all the available `FluentUISystemIcons`.
```
Comment (spencer-nelson) [inline flutter/README.md:46]: Is it worth setting 'example' as a hyperlink?

File: flutter/README.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+This generally when the app is not build properly. Please try the below steps:
+
+ 1. flutter pub get
+ 2. flutter clean
+ 3. Invalidate caches and restart Android Studio
+
+## Issues
+
+If you find a bug or want a feature, but don't know how to fix/implement it, please fill an [issue](https://github.com/microsoft/fluentui-system-icons/issues).
```
Comment (spencer-nelson) [inline flutter/README.md:73]: Thank you for mentioning the issues. Very helpful.

File: flutter/example/pubspec.yaml [hunk 9]
```diff
@@ -0,0 +1,78 @@
+name: flutter_sample_showcase
+description: A new Flutter project.
```
Comment (spencer-nelson) [inline flutter/example/pubspec.yaml:2]: We should probably update the description to match the functionality, something along the lines of: "An example of a Flutter project using Fluent System Icons" or something similar.

File: flutter/pubspec.yaml [hunk 10]
```diff
@@ -0,0 +1,32 @@
+name: fluentui_system_icons
+description: A new Flutter package project.
```
Comment (spencer-nelson) [inline flutter/pubspec.yaml:2]: Similar description change

File: importer/package.json [hunk 11]
```diff
@@ -18,6 +28,8 @@
   },
   "dependencies": {
     "avocado": "1.0.0",
-    "replace": "1.1.4"
+    "icon-font-generator": "^2.1.10",
```
Comment (spencer-nelson) [inline importer/package.json:31]: Has there been, or do we need, a legal review of icon-font-generator, along the lines of OSS checks?

File: flutter/LICENSE [hunk 12]
```diff
@@ -0,0 +1,22 @@
+Fluent Mobile Icons
```
Comment (spencer-nelson) [inline flutter/LICENSE:1]: Is this file necessary? We don't have it in the iOS/Android libraries.

File: flutter/README.md (hunk 6 above)
Comment (willhou) [reply flutter/README.md]: This should be "Documentation" to match what Android/iOS have.

File: flutter/README.md (hunk 7 above)
Comment (willhou) [reply flutter/README.md]: Yep, updated in b850334

File: importer/package.json (hunk 11 above)
Comment (willhou) [reply importer/package.json:31]: Happy to do that, please refer to our conversation on Teams.

File: flutter/LICENSE (hunk 12 above)
Comment (willhou) [reply flutter/LICENSE]: Not needed, removed in 0e891dd

File: flutter/example/pubspec.yaml (hunk 9 above)
Comment (willhou) [reply flutter/example/pubspec.yaml]: I agree, updated in 44303dd

File: flutter/pubspec.yaml (hunk 10 above)
Comment (willhou) [reply flutter/pubspec.yaml]: I agree, updated in 44303dd

File: .github/workflows/publish.yml (hunk 5 above)
Comment (willhou) [reply .github/workflows/publish.yml]: We are going to do a follow up PR for this right after, I updated it to point to the flutter issue: 4e02abb

File: flutter/CHANGELOG.md (hunk 1 above)
Comment (aakash1313) [reply flutter/CHANGELOG.md]: @willhou : This is also an important file , just added it back, let me know if the content is correct.  https://dart.dev/tools/pub/publishing#important-files

**Review by willhou — APPROVED** [review]
